File: LeetCode_Contest/LeetCode_Contest158/problemA.cpp
#include <bits/stdc++.h>
#include <string>
#include <set>

using namespace std;

class Solution {
public:
    int balancedStringSplit(string s) {
        int ans = 0, cnt = 0;
        for(auto ch : s) {
            if(ch == 'R') cnt++;
            else cnt--;
            ans += !cnt;
        }
        return ans;
    }
};

int main() {
    Solution e;
    string t;
    cin >> t;
    cout << e.balancedStringSplit(t) << endl;
    return 0;
}

File: LeetCode_Contest/LeetCode_Contest158/problemA.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        cnt , ans = 0, 0
        for c in s:
            if c == 'R':
                cnt = cnt + 1
            else :
                cnt = cnt - 1
            if cnt == 0:
                ans = ans + 1
        return ans

e = Solution()
str = input('')
print('%d' % e.balancedStringSplit(str))